File: core/src/com/mygdx/game/runtime/PointerState.cs
using Client.Domain;
using System.Diagnostics;

namespace Client.Runtime
{
    /// <summary>
    /// Contains data related to pointer (e.g. mouse pointer on desktop app) known location and 
    /// pointer command. 
    /// 
    /// To implement that abstract Pointer state we need to translate every platform-dependatn 
    /// input operation as well-known pointer command like 'select', 'zoom' etc.
    /// </summary>
    [DebuggerDisplay("Position: X:{Position.X} Y:{Position.Y}")]
    public record PointerState
    {
        public GeoPoint Position { get; init; }
    }
}

File: core/src/com/mygdx/game/runtime/PointerState.java
package com.mygdx.game.runtime;

import com.mygdx.game.domain.GeoPoint;

import lombok.Value;

/**
 * Contains data related to pointer (e.g. mouse pointer on desktop app) known location and 
 * pointer command. 
 * <p>
 * To implement that abstract Pointer state we need to translate every platform-dependatn 
 * input operation as well-known pointer command like 'select', 'zoom' etc.
 */
@Value
public class PointerState {
    private GeoPoint position;
}

